### PR TITLE
Weightless movement refactor

### PR DESCRIPTION
**Pull request by metalgearsloth**
Rather than hard stopping at walls there's now a more "realistic" motion involved. It's also not predicted because predicting physics is ASS and given it's impulse based you can't tell at "realistic" pings (AKA sub 150ms). The other upside is that if it works well enough jetpacks can probably also be implemented similarly, + grappling hooks.

Potentially more controversial change so may be closed.

tl;dr it plays more like rain world.

:cl:
- tweak: Weightless movement has been reworked.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Will re-open when I get time to work on jetpacks again but the prototype is there.
